torch_frame/datasets/data_frame_text_benchmark.py

```diff
@@ -430,14 +432,31 @@ def __init__(
 
         # Add split col
         df = dataset.df
-        if SPLIT_COL in df.columns:
-            df.drop(columns=[SPLIT_COL], inplace=True)
-        split_df = pd.DataFrame({
-            SPLIT_COL:
-            generate_random_split(length=len(df), seed=split_random_state,
-                                  train_ratio=0.8, val_ratio=0.1)
-        })
-        df = pd.concat([df, split_df], axis=1)
+        if class_name != 'MultimodalTextBenchmark':
+            if SPLIT_COL in df.columns:
+                df.drop(columns=[SPLIT_COL], inplace=True)
+            split_df = pd.DataFrame({
+                SPLIT_COL:
+                generate_random_split(length=len(df), seed=split_random_state,
+                                      train_ratio=0.8, val_ratio=0.1)
+            })
+            df = pd.concat([df, split_df], axis=1)
+        else:
+            # Manually split validation set from the train one:
+            df = df.sort_values(by=[SPLIT_COL])
+            if len(df[SPLIT_COL].unique()) == 2:
+                ser = df[SPLIT_COL]
+                train_ser = ser[ser == SPLIT_TO_NUM['train']]
+                split_ser = generate_random_split(length=len(train_ser),
+                                                  seed=split_random_state,
+                                                  train_ratio=0.9,
+                                                  val_ratio=0.1,
+                                                  include_test=False)
+                split_ser = np.concatenate([
+                    split_ser,
+                    np.full(len(df) - len(split_ser), SPLIT_TO_NUM['test'])
+                ])
+                df[SPLIT_COL] = split_ser
```

Comment:
Can you add an else here? Maybe an exception or something, just to avoid silent failures if there are not exactly 2 unique split values?